plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/action/UpdateQueryGroupRequest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.action;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.cluster.metadata.QueryGroup;
+import org.opensearch.cluster.metadata.QueryGroup.ResiliencyMode;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.search.ResourceType;
+import org.joda.time.Instant;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A request for update QueryGroup
+ *
+ * @opensearch.experimental
+ */
+public class UpdateQueryGroupRequest extends ActionRequest {
+    private final String name;
+    private final Map<ResourceType, Double> resourceLimits;
+    private final ResiliencyMode resiliencyMode;
```

Comment:
Ideally we should read it from QueryGroup instead of redefining subset of fields here again!

Reply:
I think this will be incorrect
* QueryGroup is an entity with all fields as mandatory. Pushing the partial state logic seems out of place
* This is a separate entity. if there are new fields that are added in the future to query group, we will have to make the corresponding changes in the client api code as well.
* Currently for example entities like `SearchRequest` also have evolved over time and we always add fields in them.

Although this could be achieved with something like below
```
class QueryGroup {
      // Mandatory field
      ModifiableFieldObject partialQueryGroup;
}

class ModifiableFieldObject {
      mode;
      resourceLimits;
 ...
      // do field level ser/de
 ...
      // field level getters
 ...
}
```

Reply:
> QueryGroup is an entity with all fields as mandatory. Pushing the partial state logic seems out of place

I don't think we need to push it out, just compose it in a manner that gives us flexibility with the APIs

> Currently for example entities like SearchRequest also have evolved over time and we always add fields in them.

But, do we also introduce the same field at multiple other places in addition to SearchRequest?

> This is a separate entity. if there are new fields that are added in the future to query group, we will have to make the corresponding changes in the client api code as well.

Why do we need to change the client api code? It should automatically be generally from the API spec? Unless, I am misunderstanding your concern here.

ModifiableFieldObject seems like a good way to me for reusing the definition and ser/deser logic across `QueryGroup` and `UpdateQueryGroupRequest`

Reply:
Considering a corresponding UpdateQueryGroupRequest will also be defined on the client side of the code [here](https://github.com/opensearch-project/opensearch-java/tree/main), I wonder if it is a good idea to include a queryGroup field here as it is not meant to be exposed outside. 
If we include queryGroup here, then we will need some mechanism to read the incoming request and form a queryGroup object? In which case keeping these fields as it is might be fine.

I personally think keeping UpdateQueryGroupRequest object inside core and client side might be simple and better?